src/averageAges.js

```diff
@@ -36,8 +35,13 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+
+function calculateWomenAverageAge(people, withChildren = false) {
+  const womenArray = (withChildren === false)
+    ? people.filter(woman => woman.sex === 'f')
+    : people.filter(woman => people.map(w => w.mother).includes(woman.name));
```

Comment:
Use filter just once and add some conditions to callback.
Do not use data types in variable names

```suggestion
  const women = people.filter(person => person.sex === 'f'
    && (withChildren
      ? people.some(child => child.mother === person.name)
      : true)
  );
```